operator/src/main/java/org/keycloak/operator/Config.java

```diff
@@ -27,7 +30,32 @@ public interface Config {
     Keycloak keycloak();
 
     interface Keycloak {
-        String image();
+        Image image();
         String imagePullPolicy();
+
+        interface Image {
+            String name();
+            Optional<String> tag();
+
+            default String getFinalTag() {
```

Comment:
I wonder how to test this, i.e. how much can we trust `quarkus.application.version`.

But I think we should at least have a unit test for this. @andreaTP Can you please add one?